workspaces/local-cli/src/shared/coverage-2.ts

```diff
@@ -0,0 +1,129 @@
+//TODO: Figure out where this logic should go
+
+import { IPathMapping } from "@useoptic/cli-config";
+import { getSpecEventsFrom } from "@useoptic/cli-config/build/helpers/read-specification-json";
+import { CaptureInteractionIterator } from "@useoptic/cli-shared/build/captures/avro/file-system/interaction-iterator";
+import { IHttpInteraction } from "@useoptic/domain-types";
+import * as DiffEngine from '@useoptic/diff-engine-wasm/engine/build';
+
+// This concept came from the Scala codebase's "Concern" concept
+// https://github.com/opticdev/optic-core/blob/master/core/optic/shared/src/main/scala/com/useoptic/coverage/CoverageConcerns.scala
+// JS/TS doesn't have the kind of value-types or case-classes that scala does,
+// but we can emulate them like this. I think...
+
+// o.O
+const createDataClass = <P extends Array<any>>(func: (...args: P)=>void): (...args: P)=>string => {
+  return (...args: P) => `${func.name}-${args.map(arg=>JSON.stringify(arg)).join("-")}`
```

Comment:
did you get this from someplace in particular? I think it would be better to JSON.stringify(args) than to stringify each one separately. but looks fine either way

Reply:
I make it up. Can just `JSON.stringify` and be done with, it just feels... weird to use JSON strings as keys in a map.

Reply:
I like the combo of function.name as namespace + serializing its args. ideally it would also keep the keys, which is more evolvable, but certainly fine if it's an isolated report that doesn't get transferred anywhere. anything else you do would be making up some random serialization algorithm so it's safest to stick with json

Reply:
Fair enough. I was thinking about evolvability and versions and stuff, but with the keys that's kind of a moot point. Will change

Reply:
Done!